### PR TITLE
Tag datasource: Assign tag to new note

### LINKED ISSUE #607: a new note to a tag based project is not part of the project
### What would you like to be added?

When adding a new note to a tag based project the note does not open for editing and because the new empty note does not have the project tag the new note will not show up in the project at all.  An option to allow opening a note created with the button in projects would be the simplest, automatically adding the projects tag to the front-matter would be ideal.

Work around:  Create a new set of templates for each project which includes the tag or use Templater plugin and have some sort of prompt for adding the tag  in every template.  Neither is ideal.

### Why is this needed?

Very frustrating to create a new note from within a tag based project and have nothing seem to happen and no way to easily access the new note from the current project.

### PR DESCRIPTION
**Pull request by Acylation**
fixes #607 

Don't know if this is already in progress, I would propose this anyway.

### CODE REVIEW

File: src/lib/dataApi.ts [hunk 1]
```diff
@@ -209,6 +209,13 @@ export function createDataRecord(
     path = project.newNotesFolder;
   }
 
+  if (project.dataSource.kind == "tag") {
+    values = {
+      ...values,
+      ["tags"]: [project.dataSource.config.tag.replace("#", "")],
```
Comment (Acylation) [inline src/lib/dataApi.ts:215]: Adapt to the new `Properties` feature, we can only accept tags by default.

image

**Comment by Acylation** [comment]
Currently, if the user template contains `tags` field, code here will simply override the predefined tags. If we want to fix this, we can hack into the `createNote()` function, where the template applies.

https://github.com/marcusolsson/obsidian-projects/blob/42bef466714d48ecf336c3e974fd275341490d7c/src/lib/dataApi.ts#L72-L77

**Comment by MaxwellWibert** [comment]
This might be a bit hacky, but you could pull the yaml frontmatter directly out of the interpolated content inside the createNote method. Something like as follows 

```ts
import { decodeFrontMatte, encodeFrontMatter } from './metadata'
...
  async createNote(record: DataRecord, templatePath: string): Promise<void> {
...
        content = interpolateTemplate(content, {
          title: () => getNameFromPath(record.id),
          date: (format) => moment().format(format || "YYYY-MM-DD"),
          time: (format) => moment().format(format || "HH:mm"),
        });
        if (project.datasource.kind == "tag") {
          frontMatter = decodeFrontMatter(content)
          projectTag = record.values.tags[0]
          if (frontMatter.tags && !frontMatter.tags.includes(projectTag) {
            frontMatter.tags.append(projectTag)
          }
          content = encodeFrontMatter(content, frontMatter, "plain")
        }
...
   }
```

File: src/lib/dataApi.ts [hunk 2]
```diff
@@ -75,6 +75,17 @@ export class DataApi {
...
+        if (record.values["tags"]) {
+          const templateTags = F.pipe(
+            content,
+            decodeFrontMatter,
+            E.map((frontmatter) => frontmatter["tags"]),
+            E.getOrElse(() => [])
+          );
+          //@ts-ignore explict input in `createDataRecord()`
+          const tagSet = new Set(templateTags.concat(record.values["tags"]));
+          record.values["tags"] = [...tagSet];
```
Comment (Acylation) [inline src/lib/dataApi.ts:87]: Get rid of duplicated tag value.

**Pull request merged**